### ISSUE TITLE
Error on 'npm install' - Unexpected key 'output' found

### ISSUE THREAD

**Issue by gavinr** [Esri/esri-leaflet-heatmap#21]
When I run `npm install` in this repository, I get an error on the `build` portion:

```
> esri-leaflet-heatmap@2.0.0 build C:\.....\esri-leaflet-heatmap
> rollup -c profiles/debug.js & rollup -c profiles/production.js

Unexpected key 'output' found, expected one of: banner, dest, entry, exports, external, footer, format, globals, indent, intro, moduleId, moduleName, onwarn, outro, plugins, sourceMap
Error: Unexpected key 'output' found, expected one of: banner, dest, entry, exports, external, footer, format, globals, indent, intro, moduleId, moduleName, onwarn, outro, plugins, sourceMap
    at validateKeys (C:\.....\esri-leaflet-heatmap\node_modules\rollup\www\ROLLUP\rollup\src\utils\validateKeys.js:12:11)
    at Object.rollup (C:\.....\esri-leaflet-heatmap\node_modules\rollup\www\ROLLUP\rollup\src\rollup.js:39:16)
    at bundle (C:\.....\esri-leaflet-heatmap\node_modules\rollup\bin\runRollup.js:136:16)
    at execute (C:\.....\esri-leaflet-heatmap\node_modules\rollup\bin\runRollup.js:125:3)
    at C:\.....\esri-leaflet-heatmap\node_modules\rollup\bin\runRollup.js:69:4
    at tryCatch (C:\.....\esri-leaflet-heatmap\node_modules\rollup\www\ROLLUP\rollup\node_modules\es6-promise\lib\es6-promise\-internal.js:185:12)
    at invokeCallback (C:\.....\esri-leaflet-heatmap\node_modules\rollup\www\ROLLUP\rollup\node_modules\es6-promise\lib\es6-promise\-internal.js:197:13)
    at publish (C:\.....\esri-leaflet-heatmap\node_modules\rollup\www\ROLLUP\rollup\node_modules\es6-promise\lib\es6-promise\-internal.js:168:7)
    at flush (C:\.....\esri-leaflet-heatmap\node_modules\rollup\www\ROLLUP\rollup\node_modules\es6-promise\lib\es6-promise\asap.js:87:5)
    at process._tickCallback (internal/process/next_tick.js:61:11)
Type rollup --help for help, or visit https://github.com/rollup/rollup/wiki
```

I think this is pointing to a general problem with some of our dependencies being out of date. But also we should probably do a general update of our entire `package.json`:

(I'm happy to move these into separate issues, mostly looking for feedback here @jgravois @patrickarlt)

## 1. prepublish

NPM is currently warning 

```
npm WARN prepublish-on-install 
As of npm@5, `prepublish` scripts are deprecated. 
Use `prepare` for build steps and `prepublishOnly` for upload-only. 
See the deprecation note in `npm help scripts` for more information.
```

Should we switch to `prepare` per their advice?

## 2. Update `dependencies`

- Update `esri-leaflet` to latest (v2.3.0)
- `leaflet` is currently at `^1.0.0-rc.3` - should we update this to latest (`v1.5.1`)?

## 3. Update `devDependencies`
On a cursory glance, the [`devDependencies` in our package.json](https://github.com/Esri/esri-leaflet-heatmap/blob/30d94e743555b5a7e297ce8670addc00e6ed151d/package.json#L15-L40) for this project are quite out-of-date. Major focus candidates would be:

-  rollup (currently `0.25.4`, latest: `v1.22.0`)
- karma (currently: `^0.12.24`, latest: `4.3.0`)
- chai (currently: `2.3.0`, latest: `4.2.0`
- `istanbul` - re-named to `nyc`?

## 4. jspm

Is [having the `jspm` property in our package.json](https://github.com/Esri/esri-leaflet-heatmap/blob/30d94e743555b5a7e297ce8670addc00e6ed151d/package.json#L43-L46) still the best way to do this? I'm not super familiar with jspm but looks like there have been some major updates (v0.17 and now v2.0.0-beta) since 2016 so wondering if our pattern here needs to be updated.

**Comment by jgravois** [comment]
> When I run `npm install` in this repository, I get an error

you are already on the right track. the error is caused by the fact that this plugin imports a base [rollup.config](https://github.com/Esri/esri-leaflet-heatmap/blob/94384e78f8d8cdf6c0dd4f35e487b509165a8c55/profiles/base.js#L1) from Esri-leaflet, which was upgraded to Rollup `1.x` quite awhile ago.

> Should we switch to `prepare`

yes! this is another thing I did in Esri-leaflet and didn't bother to make consistent in all the other plugins

> Update `dependencies`

The dependencies shouldn't be 'up to date', they should advertise the 'lowest' compatible version (ie: leaflet 1.0.0). see https://github.com/Esri/arcgis-rest-js/pull/506 for more info.

 > Update `devDependencies`

that's certainly housekeeping that would be helpful. if it were me I'd be spending time getting Esri-leaflet on the most modern testing stack possible before I worried about upgrading any of the other plugins though.

>  jspm

no clue! I've never used jspm.
